Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/autoClasses/League1_AutoFramework.java

```diff
@@ -0,0 +1,201 @@
+package org.firstinspires.ftc.team6220_CENTERSTAGE.autoClasses;
+
+import com.acmerobotics.dashboard.config.Config;
+import com.qualcomm.hardware.rev.RevHubOrientationOnRobot;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.DcMotorEx;
+import com.qualcomm.robotcore.hardware.IMU;
+
+import org.firstinspires.ftc.team6220_CENTERSTAGE.Utilities;
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+
+@Config
+
+abstract public class League1_AutoFramework extends LinearOpMode {
+
+    public enum AutoAlliance {
+        BLUE,
+        RED
+    }
+    public enum AutoType {
+        SHORT,
+        LONG
+    }
+
+    public void doAutoDriveInches(AutoAlliance autoAlliance, AutoType autoType) throws InterruptedException {
+
+        initHardware();
+
+        waitForStart();
+
+        // strafe right on red is +x, -x on blue
+        double strafeSignFlip = autoAlliance == AutoAlliance.BLUE ? -1 : 1;
+
+        switch (autoType) {
+
+            case SHORT:
+
+                driveInches(0, 24 * 2); // drive forward to parking spot
+
+                break;
+
+            case LONG:
+
+                sleep(15000); // wait for 15 seconds so alliance an finish
+
+                driveInches(-3 * strafeSignFlip, 0); // strafe away from the wall 3 inches
+                driveInches(0, 24 * 4); // drive forward to parking spot
+
+                break;
+        }
+
+    }
+
+
+    static final double TICKS_PER_REVOLUTION = 537.7;
```

Comment:
Put into constants class please :)

Reply:
The TICKS_PER_REVOLUTION constant value is not correct.